src/components/markdown/ResourceStatusCheckbox.tsx

```diff
@@ -6,6 +6,7 @@ import 'tippy.js/themes/light.css';
 import ConfettiContext from '../../context/ConfettiContext';
 import { useDarkMode } from '../../context/DarkModeContext';
 import MarkdownLayoutContext from '../../context/MarkdownLayoutContext';
+import { replaceIllegalFirebaseCharacters } from '../../context/UserDataContext/properties/userProgressOnResources';
```

Comment:
mind explaining the changes you made?

Reply:
I read the problem progress from local storage every time before writing to Firebase, changed `setDoc` to `updateDoc`, and used `arrayUnion` to add an element into the progress array rather than override the entire array. Reasoning for each change:

- local storage: this shouldn't do anything, especially after the other two changes. (In fact I am like pretty sure this change does nothing)
- setDoc => updateDoc: this allows us to use `arrayUnion`. It's possible that a poor internet connection could cause user activity array to be overwritten by an outdated user activity array (this would be fixed by arrayUnion), but this doesn't explain why user problem progress is being lost.

Reply:
ok, then should we leave the issue open?

Reply:
Probably. Maybe we can make one mega issue for reporting lost data and pin it?

Reply:
ok